ipapython/dn.py

```diff
@@ -1419,6 +1419,15 @@ def rindex(self, pattern, start=None, end=None):
             raise ValueError("pattern not found")
         return i
 
+    def get_reverse(self):
+        '''
+        Return a reverse copy of the DN. For instance, the reverse of
+        uid=user,dc=example,dc=com is dc=com,dc=example,uid=user
+        '''
+        reverse_dn = DN(self)
+        reverse_dn.rdns.reverse()
+        return reverse_dn
```

Comment:
There is no such thing as reversed DN - it's only the the textural representation which might be in both directions, but the underlying DN is the same. I would prefer if you instead added two new methods `ldap_text()` and `x500_text()` which return textual representation of the DN in the LDAP and X.500 order, respectively, and use them as appropriate:
```python
    def ldap_text(self):
        return dn2str([[self.to_openldap()]])

    def x500_text(self):
        return dn2str([[reversed(self.to_openldap()]])

    def __str__(self):
        return self.ldap_text()
```